bot/exts/filtering/filtering.py

```diff
@@ -0,0 +1,1424 @@
+            ctx.author,
+            embed,
+            patch_func
+        )
+        await ctx.send(embed=embed, reference=ctx.message, view=view)
+
+    @filter.command(name="delete", aliases=("d", "remove"))
+    async def f_delete(self, ctx: Context, filter_id: int) -> None:
+        """Delete the filter specified by its ID."""
+        async def delete_list() -> None:
+            """The actual removal routine."""
+            await bot.instance.api_client.delete(f'bot/filter/filters/{filter_id}')
+            log.info(f"Successfully deleted filter with ID {filter_id}.")
+            filter_list[list_type].filters.pop(filter_id)
+            await ctx.reply(f"✅ Deleted filter: {filter_}")
+
+        result = self._get_filter_by_id(filter_id)
+        if result is None:
+            await ctx.send(f":x: Could not find a filter with ID `{filter_id}`.")
+            return
+        filter_, filter_list, list_type = result
+        await ctx.reply(
+            f"Are you sure you want to delete filter {filter_}?",
+            view=DeleteConfirmationView(ctx.author, delete_list)
+        )
+
+    @filter.command(aliases=("settings",))
+    async def setting(self, ctx: Context, setting_name: str | None) -> None:
+        """Show a description of the specified setting, or a list of possible settings if no name is specified."""
+        if not setting_name:
+            settings_list = [f"» {setting_name}" for setting_name in self.loaded_settings]
+            for filter_name, filter_settings in self.loaded_filter_settings.items():
+                settings_list.extend(f"» {filter_name}/{setting}" for setting in filter_settings)
+            embed = Embed(colour=Colour.blue())
+            embed.set_author(name="List of setting names")
+            await LinePaginator.paginate(settings_list, ctx, embed, max_lines=10, empty=False)
+
+        else:
+            # The setting is either in a SettingsEntry subclass, or a pydantic model.
+            setting_data = self.loaded_settings.get(setting_name)
+            description = None
+            if setting_data:
+                description = setting_data[0]
+            elif "/" in setting_name:  # It's a filter specific setting.
+                filter_name, filter_setting_name = setting_name.split("/", maxsplit=1)
+                if filter_name in self.loaded_filter_settings:
+                    if filter_setting_name in self.loaded_filter_settings[filter_name]:
+                        description = self.loaded_filter_settings[filter_name][filter_setting_name][0]
+            if description is None:
+                await ctx.send(f":x: There's no setting type named {setting_name!r}.")
+                return
+            embed = Embed(colour=Colour.blue(), description=description)
+            embed.set_author(name=f"Description of the {setting_name} setting")
+            await ctx.send(embed=embed)
+
+    @filter.command(name="match")
+    async def f_match(
```

Comment:
This command doesn't seem to detect discord invites (that aren't whitelisted or blacklisted, but still trigger a notification).

Reply:
Yes, the goal is to find filters. In a previous iteration I just made it spit out the same messages that would appear in the mod alert, but this seemed more useful. Should it be reverted?

Reply:
Hmm, it probably does make more sense as-is actually, maybe the help text could be clarified though.

Reply:
[Correct filter match docstring](https://github.com/python-discord/bot/pull/2390/commits/a62993b44cfc0f2c3c5d746005b9a716dea82add)